test/libsolidity/SemanticTest.cpp

```diff
@@ -234,8 +246,23 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 			return TestResult::Failure;
 		}
 
+		if (success && !m_runWithEwasm && _compileToEwasm)
+		{
+			if (m_revertStrings != RevertStrings::Default)
+				return TestResult::Success;
+
+			m_compileToEwasmCanBeSet = true;
+			AnsiColorized(_stream, _formatted, {BOLD, YELLOW}) <<
+				_linePrefix << endl <<
+				_linePrefix << "Test can pass via Yul to ewasm and marked with compileToEwasm: false." << endl;
+			return TestResult::Failure;
+		}
+
 		if (!success && (m_runWithYul || !_compileViaYul))
 		{
+			if (!m_runWithEwasm && m_enforceCompileToEwasm)
+				return TestResult::Success;
```

Comment:
Is this just to ignore failing cases that can't yet be compiled to ewasm?

Reply:
Yes exactly

Reply:
Can you add a comment that explains this?